log4j-api/src/main/java/org/apache/logging/log4j/message/ParameterFormatter.java

```diff
@@ -448,7 +450,7 @@ static String deepToString(final Object o) {
      * @param str    the StringBuilder that o will be appended to
      * @param dejaVu a list of container identities that were already used.
      */
-    static void recursiveDeepToString(final Object o, final StringBuilder str, final Set<String> dejaVu) {
+    static void recursiveDeepToString(final Object o, final StringBuilder str, final Set<Object> dejaVu) {
```

Comment:
Would it make sense to make this method private and instead add a package-private overload `recursiveDeepToString(Object, StringBuilder)`?
Most of the calls to this method have `null` as dejaVu, and especially the calls from `MapMessage` should not have to care about how recursion tracking is implemented.

Reply:
Makes sense, implemented.